src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/AddressDto.java

```diff
@@ -137,4 +139,25 @@ public String getPostcode() {
     public void setPostcode(String postcode) {
         this.postcode = postcode;
     }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final AddressDto addressDto = (AddressDto) o;
+        return Objects.equals(line1, addressDto.line1) &&
+                Objects.equals(line2, addressDto.line2) &&
+                Objects.equals(careOf, addressDto.careOf) &&
+                Objects.equals(country, addressDto.country) &&
+                Objects.equals(locality, addressDto.locality) &&
+                Objects.equals(poBox, addressDto.poBox) &&
+                Objects.equals(postcode, addressDto.postcode) &&
+                Objects.equals(propertyNameNumber, addressDto.propertyNameNumber) &&
+                Objects.equals(county, addressDto.county) &&
+                Objects.equals(town, addressDto.town);
+    }
```

Comment:
Usually the HashCode() method would be overridden in conjunction with the equals() method. IntelliJ does this for you if you right-click anywhere in a class and select Generate, the option for generating equals & hashcode appears.